crates/router/src/connector/payone.rs

```diff
@@ -105,33 +160,53 @@ impl ConnectorCommon for Payone {
         res: Response,
         event_builder: Option<&mut ConnectorEvent>,
     ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
-        let response: payone::PayoneErrorResponse = res
+        let response: payone::ErrorResponse = res
             .response
-            .parse_struct("PayoneErrorResponse")
+            .parse_struct("ErrorResponse")
             .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
 
         event_builder.map(|i| i.set_response_body(&response));
         router_env::logger::info!(connector_response=?response);
 
-        Ok(ErrorResponse {
-            status_code: res.status_code,
-            code: response.code,
-            message: response.message,
-            reason: response.reason,
-            attempt_status: None,
-            connector_transaction_id: None,
-        })
+        match response.errors {
+            Some(errors) => {
+                let first_error = errors.first();
+                let code = first_error.map(|error| error.code.clone());
+                Ok(ErrorResponse {
+                    status_code: res.status_code,
+                    code: code.unwrap_or_else(|| consts::NO_ERROR_CODE.to_string()),
```

Comment:
Discuss with @sai-harsha-vardhan , how retries work, and then decide how the error_code should be populated